src/options-resolver.ts

```diff
@@ -15,33 +15,66 @@
  * limitations under the License.
  *
  */
+import chalk from 'chalk';
+import * as prompts from 'prompts';
+import { Choice, PromptType } from 'prompts';
 
-import * as readlineSync from 'readline-sync';
+export const EXIT_FLAG = '@EXIT';
 
-export const OptionsResolver = (options: any,
-                                key: string,
-                                secondSource: () => string | undefined,
-                                promptText: string,
-                                readlineDependency?: any,
-                                hide?: boolean) => {
-    const readline = readlineDependency || readlineSync;
-    const hideEchoBack = hide ? true : false;
-    return options[key] !== undefined ? options[key] : (secondSource() ||
-        readline.question(promptText, {hideEchoBack}));
+export const OptionsChoiceResolver = async (options: any,
+                                            key: string,
+                                            promptText: string,
+                                            choices: Choice[],
+                                            type: PromptType = 'select') => {
+    if (!['select', 'multiselect'].includes(type)) {
+        console.log(chalk.red('ERR'), 'Invalid options choice resolver type');
+        return process.exit();
+    }
+    if (options[key] !== undefined) {
+        return options[key];
+    }
+    const response = await prompts({
+        type,
+        name: key,
+        message: promptText,
+        choices,
+    });
+    if (response[key] === undefined) {
+        return process.exit();
+    }
+
+    if (('select' === type && EXIT_FLAG === response[key]) || ('multiselect' === type && response[key].includes(EXIT_FLAG))) {
+        return process.exit();
+    }
+    return response[key];
 };
 
-export const OptionsChoiceResolver = (options: any,
+export const OptionsResolver = async (options: any,
                                       key: string,
+                                      secondSource: () => string | undefined,
                                       promptText: string,
-                                      choices: string[],
-                                      readlineDependency?: any) => {
-    const readline = readlineDependency || readlineSync;
-    if (options[key] !== undefined) {
-        return options[key];
-    }
-    const choiceIndex = readline.keyInSelect(choices, promptText);
-    if (-1 === choiceIndex) {
+                                      type: PromptType = 'text') => {
```

Comment:
I would prefer to use a custom enum for ``type`` instead of depending directly from the library.